tasties_app/views.py

```diff
@@ -61,3 +61,14 @@ def register(request):
 
     context = {'form': form}
     return render(request, 'tasties_app/register.html', context)
+
+
+@login_required(login_url='login')
+def single_recipe(request, recipe=None):
+    if not recipe:
+        recipe = Recipe.objects.first()
+    ingredients = Ingredient.objects.filter(recipe_id=recipe)
+    rating = Rating.objects.filter(recipe_id=recipe).aggregate(Avg('rating'))['rating__avg']
```

Comment:
Here you can definitely utilize `rating = recipe.rating_set.aggregate(Avg('rating__rating'))`